src/core/main.c

```diff
@@ -1528,6 +1528,12 @@ int main(int argc, char *argv[]) {
                  * need to do that for user instances since they never log
                  * into the console. */
                 log_show_color(colors_enabled());
+                /* make_null_stdio would cause /dev/console to be closed in a (docker) container
+                 * journald and other processes would not work properly anymore
+                 * if they try to log to console */
+                if (detect_container() <= 0)
+                        make_null_stdio();
+
                 make_null_stdio();
```

Comment:
Your patch does not actually remove the call to `make_null_stdio`. Did you test some different patch maybe?

Anyway, this seems extremely brittle: /dev/console should be a able to be opened and closed as many times as necessary. I don't understand why docker would delete the file, even if systemd closes it on its end.